src/metabase/pulse/render/common.clj

```diff
@@ -25,55 +28,110 @@
                                                                  (:type/Number global-settings)
                                                                  column-settings)
-
-
-        [decimal grouping] (or number-separators ".,")
+        integral?       (isa? (or effective_type base_type) :type/Integer)
+        percent?        (= number-style "percent")
+        scientific?     (= number-style "scientific")
+        [decimal grouping] (or number-separators
+                               (get-in (public-settings/custom-formatting) [:type/Number :number_separators])
+                               ".,")
         symbols            (doto (DecimalFormatSymbols.)
                              (cond-> decimal (.setDecimalSeparator decimal))
                              (cond-> grouping (.setGroupingSeparator grouping)))
-        base               (if (= number-style "scientific") "0" "#,###")
-        integral?          (isa? (or effective_type base_type) :type/Integer)
-        fmt-str            (cond-> (cond
-                                     decimals  (if (zero? decimals)
-                                                 base
-                                                 (apply str base "." (repeat decimals "0")))
-                                     integral? base
-                                     :else     (str base ".00"))
-                             (= number-style "scientific") (str "E0")
-                             (= number-style "percent")    (str "%"))
-        fmtr               (DecimalFormat. fmt-str symbols)]
+        base               (cond-> (if (= number-style "scientific") "0" "#,##0")
+                             (not grouping) (str/replace #"," ""))]
     (fn [value]
       (if (number? value)
-        (NumericWrapper.
-         (str (when prefix prefix)
-              (when (and currency? (or (nil? currency-style)
-                                       (= currency-style "symbol")))
-                (get-in currency/currency [(keyword (or currency "USD")) :symbol]))
-              (when (and currency? (= currency-style "code"))
-                (str (get-in currency/currency [(keyword (or currency "USD")) :code]) \space))
-              (.format fmtr (* value (or scale 1)))
-              (when (and currency? (= currency-style "name"))
-                (str \space (get-in currency/currency [(keyword (or currency "USD")) :name_plural])))
-              (when suffix suffix)))
+        (let [scaled-value (* value (or scale 1))
+              decimals-in-value (digits-after-decimal (if (= number-style "percent")
+                                                        (* 100 scaled-value)
+                                                        scaled-value))
+              decimal-digits (cond
+                               integral? 0
+                               currency? (get-in currency/currency [(keyword (or currency "USD")) :decimal_digits])
```

Comment:
hard to read the order, but when in currency, the explicit choice of number of decimal digits should win, and fall back the currency's default otherwise.

Reply:
ok on rereading, the `decimals` clause should be at top. If a human told us to use `n` decimal places, we should do that.